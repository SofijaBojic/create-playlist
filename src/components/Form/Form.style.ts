import styled from "styled-components";

export const FormWraperStyle: any = styled.form`
  padding: 20px;
  border: 1px solid #04274f;
  text-align: center;
  margin: 30px 0px;

  & input {
    width: 30%;
    padding: 15px;
    border: 0px;
    margin-right: 3%;
    margin-bottom: 15px;
  }
`;

export const FormButtonStyle: any = styled.button`
  padding: 10px;
  width: 70%;
  border: 0px;
  margin-bottom: 15px;
  border-radius: 23px;
`;
