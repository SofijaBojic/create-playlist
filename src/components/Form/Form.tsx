import React, { Component } from "react";
import { FormWraperStyle, FormButtonStyle } from "../Form/Form.style";
import { Grid, GridElement } from "../../styles/grid";
import { Heading1 } from "../Text/Text.style";

class Form extends Component<IFormProps, IFormState> {
  initState: IFormState = {
    file: "",
    image: "",
    title: ""
  }
  state: IFormState = this.initState;

  handleClick = (event: any): void => {
    event.preventDefault && event.preventDefault();
    const { file, image, title } = this.state;
    if (file !== "" && image !== "" && title !== "") {
      this.props.addVideo({ file, image, title });
      this.setState(this.initState);
    }
  };

  handleInputChange = (event: { target: { value: string, name: string } }): void => {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  };

  render() {
    const { file, image, title } = this.state;
    return (
      <>
        <FormWraperStyle onSubmit={(event: any) => this.handleClick(event)} noValidate>
          <Grid>
            <GridElement lgColumn="1 / 13" mdColumn="1 / 13" smColumn="1 / 7">
              <Heading1>Add video to playlist</Heading1>
              <input
                data-testid="form_file"
                name="file"
                type="text"
                placeholder="FILE URL"
                value={file}
                onChange={event => this.handleInputChange(event)}
                required
              />
              <input
                data-testid="form_image"
                name="image"
                type="text"
                placeholder="IMAGE URL"
                value={image}
                onChange={event => this.handleInputChange(event)}
                required
              />
              <input
                data-testid="form_title"
                name="title"
                type="text"
                placeholder="VIDEO TITLE"
                value={title}
                onChange={event => this.handleInputChange(event)}
                required
              />
              <FormButtonStyle type="submit" data-testid="form_button">
                Click
              </FormButtonStyle>
            </GridElement>
          </Grid>
        </FormWraperStyle>
      </>
    );
  }
}

export default Form;
