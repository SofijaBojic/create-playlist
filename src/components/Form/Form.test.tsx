import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import Form from "./index";
import Theme from "../../styles/globalVariables";

describe("Button component", () => {
    const videoItemObj: IVideoListItem = {
        file: "//vr.jwplayer.com/content/AgqYcfAT/AgqYcfAT-8yQ1cYbs.mp4",
        image: "https://i.pinimg.com/originals/ca/a2/65/caa2654e79e2dc88c6a3c18e1a353452.jpg",
        title: "Caminandes VR"
    };
    it("should render and work correctly with wrong type", () => {
        const addVideo = jest.fn();
        const { getByTestId } = render(
            <ThemeProvider theme={Theme}>
                <Form addVideo={addVideo} />
            </ThemeProvider>
        ) as any;
        const form_file: HTMLElement = getByTestId("form_file");
        const form_image: HTMLElement = getByTestId("form_image");
        const form_title: HTMLElement = getByTestId("form_title");
        const form_button: HTMLElement = getByTestId("form_button");
        fireEvent.change(form_file, {
            target: { value: videoItemObj.file, name: "file" }
        });
        fireEvent.change(form_image, {
            target: { value: videoItemObj.image, name: "image" }
        });
        fireEvent.change(form_title, {
            target: { value: videoItemObj.title, name: "title" }
        });

        expect(form_file.getAttribute('value')).toBe(videoItemObj.file);
        expect(form_image.getAttribute('value')).toBe(videoItemObj.image);
        expect(form_title.getAttribute('value')).toBe(videoItemObj.title);

        fireEvent.click(form_button);

        expect(form_file.getAttribute('value')).toBe("");
        expect(form_image.getAttribute('value')).toBe("");
        expect(form_title.getAttribute('value')).toBe("");
    });

    it("should submit form with some empty input", () => {
        const addVideo = jest.fn();
        const { getByTestId } = render(
            <ThemeProvider theme={Theme}>
                <Form addVideo={addVideo} />
            </ThemeProvider>
        ) as any;
        const form_file: HTMLElement = getByTestId("form_file");
        const form_image: HTMLElement = getByTestId("form_image");
        const form_title: HTMLElement = getByTestId("form_title");
        const form_button: HTMLElement = getByTestId("form_button");

        fireEvent.change(form_file, {
            target: { value: videoItemObj.file, name: "file" }
        });
        fireEvent.change(form_image, {
            target: { value: "", name: "image" }
        });
        fireEvent.change(form_title, {
            target: { value: "", name: "title" }
        });


        expect(form_file.getAttribute('value')).toBe(videoItemObj.file);
        expect(form_image.getAttribute('value')).toBe("");
        expect(form_title.getAttribute('value')).toBe("");
        fireEvent.click(form_button);
        expect(form_file.getAttribute('value')).toBe(videoItemObj.file);
    });
});
