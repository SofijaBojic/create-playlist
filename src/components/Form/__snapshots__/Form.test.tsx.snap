// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render and work correctly with wrong type 1`] = `
<div
  class="sc-bdVaJa byomO"
>
  <div
    class="sc-htpNat csdCIQ"
  >
    <div
      class="sc-bxivhb jqGiQm"
    >
      <h1
        class="sc-ifAKCX hybose"
      >
        Add video to playlist
      </h1>
      <input
        data-testid="form_file"
        name="file"
        placeholder="FILE URL"
        type="text"
        value=""
      />
      <input
        data-testid="form_image"
        name="image"
        placeholder="IMAGE URL"
        type="text"
        value=""
      />
      <input
        data-testid="form_title"
        name="title"
        placeholder="VIDEO TITLE"
        type="text"
        value=""
      />
      <button
        class="sc-bwzfXH dQVdn"
        data-testid="form_button"
      >
        Click
      </button>
    </div>
  </div>
</div>
`;
